lib/impl/submitFormDataFunctions/processSubmissionUpdates.js

```diff
@@ -2,39 +2,59 @@ var _ = require('underscore');
 
 module.exports = function processSubmissionUpdates(formSubmission, submissionData) {
   formSubmission.formSubmittedAgainst = formSubmission.formSubmittedAgainst || {};
+
   var allFields = _.map(formSubmission.formSubmittedAgainst.pages, function(page) {
     return page.fields || [];
   });
-
   allFields = _.compact(_.flatten(allFields));
 
-  // update
+  // for all field definitions
   _.each(allFields, function(fieldDef) {
-    var updatedFieldData = _.find(submissionData.formFields, function(updatedField) {
-      return updatedField.fieldId.toString() === fieldDef._id.toString();
-    });
-
-    var existingFieldData = _.find(formSubmission.formFields, function(existingField) {
-      return existingField.fieldId.toString() === fieldDef._id.toString();
+    // check against every submitted field, there might be more fields with the same id (repeating sections)
+    _.each(submissionData.formFields, function(updatedField) {
+      var updatedFieldId = updatedField.fieldId.toString();
+      var fieldDefId = fieldDef._id.toString();
+
+      if (updatedField.fieldId.toString() === fieldDef._id.toString()) {
+        var missingSection = false;
+        var sameId = false;
+        var sameIdField = null;
+
+        var existingFieldData = _.find(formSubmission.formFields, function(existingField) {
+          var existingFieldId = existingField.fieldId.toString();
+          sameId = existingFieldId === updatedFieldId;
+          sameIdField = sameId ? existingField : null;
+          var sameSection = existingField.sectionIndex === updatedField.sectionIndex;
+          missingSection = existingField.sectionIndex === 0 && !updatedField.sectionIndex;
+
+          //exact match for field id and its section index
+          return sameId && sameSection;
+        });
+
+        if (existingFieldData) {
+          existingFieldData.fieldValues = updatedField.fieldValues ;
+          save(formSubmission, existingFieldData);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the declaration of "save" before this usage. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)